app/models/account/payment_details.rb

```diff
@@ -4,7 +4,7 @@ module Account::PaymentDetails
   extend ActiveSupport::Concern
 
   included do
-    has_one :payment_detail, -> { order id: :desc }, autosave: true
+    has_one :payment_detail, -> { order id: :desc }, autosave: true, dependent: :delete
```

Comment:
Because:
1. This way it doesn't stay in the DB after removing the account
2. It does not get destroyed before the unstore.

This is a temporary step and **does not meant to be merged**, but it is there so it doesn't stop my tests to make things work first 😄